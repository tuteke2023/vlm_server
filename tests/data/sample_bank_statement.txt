BANK OF EXAMPLE
Account Statement

Account Number: ****1234
Statement Period: January 1-31, 2025

Date | Description | Withdrawals | Deposits | Balance
------|-------------|-------------|----------|----------
01/01/2025 | Beginning Balance | | | 2,000.00
01/05/2025 | WALMART SUPERCENTER | 125.50 | | 1,874.50
01/10/2025 | DIRECT DEPOSIT - PAYROLL | | 3,000.00 | 4,874.50
01/12/2025 | SHELL GAS STATION #1234 | 75.25 | | 4,799.25
01/15/2025 | ONLINE TRANSFER TO SAVINGS | 500.00 | | 4,299.25
01/16/2025 | STARBUCKS COFFEE | 6.75 | | 4,292.50
01/18/2025 | ATM WITHDRAWAL | 200.00 | | 4,092.50
01/20/2025 | NETFLIX SUBSCRIPTION | 15.99 | | 4,076.51
01/22/2025 | AMAZON MARKETPLACE | 89.99 | | 3,986.52
01/25/2025 | DIRECT DEPOSIT - PAYROLL | | 3,000.00 | 6,986.52
01/26/2025 | UTILITY BILL PAYMENT | 156.78 | | 6,829.74
01/28/2025 | GROCERY STORE #5678 | 234.56 | | 6,595.18
01/30/2025 | RESTAURANT XYZ | 45.80 | | 6,549.38
01/31/2025 | SERVICE FEE | 12.00 | | 6,537.38

Total Withdrawals: $1,462.62
Total Deposits: $6,000.00
Ending Balance: $6,537.38