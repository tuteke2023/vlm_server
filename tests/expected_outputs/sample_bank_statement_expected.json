{
  "account_number": "****1234",
  "statement_period": "January 1-31, 2025",
  "opening_balance": 2000.00,
  "closing_balance": 6537.38,
  "total_debits": 1462.62,
  "total_credits": 6000.00,
  "transactions": [
    {
      "date": "01/05/2025",
      "description": "WALMART SUPERCENTER",
      "category": "Groceries",
      "debit": 125.50,
      "credit": 0,
      "balance": 1874.50
    },
    {
      "date": "01/10/2025",
      "description": "DIRECT DEPOSIT - PAYROLL",
      "category": "Income",
      "debit": 0,
      "credit": 3000.00,
      "balance": 4874.50
    },
    {
      "date": "01/12/2025",
      "description": "SHELL GAS STATION #1234",
      "category": "Transportation",
      "debit": 75.25,
      "credit": 0,
      "balance": 4799.25
    },
    {
      "date": "01/15/2025",
      "description": "ONLINE TRANSFER TO SAVINGS",
      "category": "Transfer",
      "debit": 500.00,
      "credit": 0,
      "balance": 4299.25
    },
    {
      "date": "01/16/2025",
      "description": "STARBUCKS COFFEE",
      "category": "Dining",
      "debit": 6.75,
      "credit": 0,
      "balance": 4292.50
    },
    {
      "date": "01/18/2025",
      "description": "ATM WITHDRAWAL",
      "category": "Cash",
      "debit": 200.00,
      "credit": 0,
      "balance": 4092.50
    },
    {
      "date": "01/20/2025",
      "description": "NETFLIX SUBSCRIPTION",
      "category": "Entertainment",
      "debit": 15.99,
      "credit": 0,
      "balance": 4076.51
    },
    {
      "date": "01/22/2025",
      "description": "AMAZON MARKETPLACE",
      "category": "Shopping",
      "debit": 89.99,
      "credit": 0,
      "balance": 3986.52
    },
    {
      "date": "01/25/2025",
      "description": "DIRECT DEPOSIT - PAYROLL",
      "category": "Income",
      "debit": 0,
      "credit": 3000.00,
      "balance": 6986.52
    },
    {
      "date": "01/26/2025",
      "description": "UTILITY BILL PAYMENT",
      "category": "Utilities",
      "debit": 156.78,
      "credit": 0,
      "balance": 6829.74
    },
    {
      "date": "01/28/2025",
      "description": "GROCERY STORE #5678",
      "category": "Groceries",
      "debit": 234.56,
      "credit": 0,
      "balance": 6595.18
    },
    {
      "date": "01/30/2025",
      "description": "RESTAURANT XYZ",
      "category": "Dining",
      "debit": 45.80,
      "credit": 0,
      "balance": 6549.38
    },
    {
      "date": "01/31/2025",
      "description": "SERVICE FEE",
      "category": "Banking",
      "debit": 12.00,
      "credit": 0,
      "balance": 6537.38
    }
  ],
  "transaction_count": 13
}